tap_stripe/client.py

```diff
@@ -321,27 +324,113 @@ def _write_state_message(self) -> None:
+        end_date = datetime.fromtimestamp(context["concurrent_params"].get("created[lt]", datetime.utcnow().timestamp())).isoformat()
+        self.logger.info(f"Fetching data concurrently from {self.name} from {start_date} to {end_date}")
+
+        while not finished:
+            prepared_request = self.prepare_request(
+                context, next_page_token=next_page_token
+            )
+            resp = decorated_request(prepared_request, context)
+            consolidated_response.extend(self.parse_response(resp))
+            previous_token = copy.deepcopy(next_page_token)
+            next_page_token = self.get_next_page_token(
+                response=resp, previous_token=previous_token
+            )
+            if next_page_token and next_page_token == previous_token:
+                raise RuntimeError(
+                    f"Loop detected in pagination. "
+                    f"Pagination token {next_page_token} is identical to prior token."
+                )
+            # Cycle until get_next_page_token() no longer returns a value
+            finished = not next_page_token
+        return consolidated_response
+    
+    def get_concurrent_params(self, context, max_requests):
+        start_date = self.get_starting_time(context)
+        start_date_timestamp = int(start_date.timestamp())
+        end_date_timestamp = int(datetime.utcnow().timestamp())
+
+        current_start = start_date_timestamp
+        default_partition_size = 2592000 # one month as default
+        partition_size = min(math.ceil((end_date_timestamp - start_date_timestamp) / max_requests), default_partition_size)
+
+        concurrent_params = []
+
+        chunks = math.ceil((end_date_timestamp - start_date_timestamp)/partition_size)
+        for i in range(chunks):
+            params = {}
+            current_end = current_start + partition_size
+            params["created[gte]"] = current_start
+            # don't put an end_date at the final chunk
+            if i < chunks-1:
+                params["created[lt]"] = current_end
+            concurrent_params.append({"concurrent_params": params})
+            current_start = current_end
+        
+        return concurrent_params
+
+    
+    def request_records(self, context: Optional[dict]) -> Iterable[dict]:
+        max_requests = 1
+        requests_params = []
+
+        # use sequential requests for incremental syncs
+        if not self.stream_state.get("replication_key_value") or not self.replication_key:
+            max_requests = 25 if "live" in self.config.get("client_secret") else 10
+            requests_params = self.get_concurrent_params(context, max_requests)
+        
+        if len(requests_params):
+            for i in range(0, len(requests_params), max_requests):
+                req_params = requests_params[i: i + max_requests]
+
+                with concurrent.futures.ThreadPoolExecutor(
```

Comment:
If we need true parallelism for CPU-bound tasks, the `concurrent.futures.ProcessPoolExecutor` is often a better choice. It uses separate processes rather than threads, allowing each to have its own Python interpreter and thus bypassing the GIL.

However, for I/O-bound tasks (like network requests, file I/O, or waiting on APIs), ThreadPoolExecutor can still be quite effective than the previous implementation.